src/block-components/icon/edit.js

```diff
@@ -0,0 +1,320 @@
+									attribute="iconColor2"
+									hover="all"
+								/>
+
+								<AdvancedRangeControl
+									label={ __( 'Gradient Direction (degrees)', i18n ) }
+									attribute="iconColorGradientDirection"
+									hover="all"
+									min={ 0 }
+									max={ 360 }
+									step={ 10 }
+									allowReset={ true }
+								/>
+							</Fragment>
+						) }
+					</Fragment>
+				) }
+
+				{ ! hasGradient && (
+					<ColorPaletteControl
+						label={ __( 'Icon Color', i18n ) }
+						attribute="iconColor1"
+						hover="all"
+					/>
+				) }
+
+				<AdvancedRangeControl
+					label={ __( 'Icon Size', i18n ) }
+					attribute="iconSize"
+					min={ 0 }
+					max={ 100 }
+					step={ 1 }
+					allowReset={ true }
+					hover="all"
+				/>
+
+				<AdvancedRangeControl
+					label={ __( 'Icon Opacity', i18n ) }
+					attribute="iconOpacity"
+					min={ 0 }
+					max={ 1 }
+					step={ 0.1 }
+					allowReset={ true }
+					placeholder="1.0"
+					hover="all"
+				/>
+
+				<AdvancedRangeControl
+					label={ __( 'Icon Rotation', i18n ) }
+					attribute="iconRotation"
+					min={ 0 }
+					max={ 360 }
+					allowReset={ true }
+					placeholder="0"
+					hover="all"
+				/>
+
+				<AdvancedSelectControl
+					label={ __( 'Icon Position', i18n ) }
+					attribute="iconPosition"
+					options={ [
+						{ value: '', label: __( 'Left', i18n ) },
+						{ value: 'right', label: __( 'Right', i18n ) },
+					] }
+				/>
+
+				<AdvancedRangeControl
+					label={ __( 'Icon Gap', i18n ) }
+					attribute="iconGap"
+					min={ 0 }
+					max={ 50 }
+					allowReset={ true }
+					placeholder="0"
+				/>
+
+				{ hasShape && (
+					<Fragment>
+						<ToggleControl
+							label={ __( 'Shaped', i18n ) }
+							checked={ getAttribute( 'shaped' ) }
+							onChange={ updateAttributeHandler( 'shaped' ) }
+						/>
+
+						{ getAttribute( 'shaped' ) && (
+							<Fragment>
+
+								<ColorPaletteControl
+									label={ __( 'Shape Color', i18n ) }
+									attribute="shapeColor"
+									hover="all"
+									hasTransparent={ true }
+								/>
+
+								<AdvancedRangeControl
+									label={ __( 'Shape Border Radius', i18n ) }
+									attribute="shapeBorderRadius"
+									hover="all"
+									min={ 0 }
+									max={ 100 }
+									step={ 1 }
+									allowReset={ true }
+									placeholder={ 50 }
+								/>
+
+								<AdvancedRangeControl
+									label={ __( 'Shape Padding', i18n ) }
+									attribute="shapePadding"
+									hover="all"
+									min={ 0 }
+									max={ 150 }
+									step={ 1 }
+									allowReset={ true }
+									placeholder={ 20 }
+								/>
+
+								<ToggleControl
+									label={ __( 'Shape Outline', i18n ) }
+									checked={ getAttribute( 'shapeOutline', 'desktop', state ) }
+									onChange={ updateAttributeHandler( 'shapeOutline', 'desktop', state ) }
+								/>
+
+								{ getAttribute( 'shapeOutline', 'desktop', state ) && (
```

Comment:
As with color types, let's keep things simple. shape outlines should be not be responsive or hoverable. the shape's color, opacity and other attributes can be changed